components/navigation/sideBar.js

```diff
@@ -1,110 +1,96 @@
-import React from "react";
-import { connectScrollTo } from "react-instantsearch-dom";
+import React, { useState, useEffect } from "react";
 import bus from "../../lib/bus";
-
 import NavItem from "../navigation/navItem";
 
-export default class SideBar extends React.Component {
-  constructor(props) {
-    super(props);
-
-    this.state = {
-      condensed: false,
-      loading: true,
-      depth: 1,
-      sticky: false,
-      over: false,
-      open: false,
-      theme: "light-mode",
-      menu: props.menu,
-    };
-
-    this.checkExpanded = this.checkExpanded.bind(this);
-    this.handleMouseEnter = this.handleMouseEnter.bind(this);
-    this.handleMouseLeave = this.handleMouseLeave.bind(this);
-    this.handleTheme = this.handleTheme.bind(this);
-  }
-
-  handleTheme() {
-    this.setState({ theme: document.body.dataset.theme });
-  }
-
-  handleScroll() {
+const SideBar = ({ menu, slug, paths, version, maxVersion }) => {
+  const [state, setState] = useState({
+    condensed: false,
+    loading: true,
+    depth: 1,
+    sticky: false,
+    over: false,
+    open: false,
+    theme: "light-mode",
+    menu,
+  });
```

Comment:
I think it's more correct to pull each of these into their own `useState`:

```
const [isCondensed, setIsCondensed] = useState(false);
const [isLoading, setIsLoading] = useState(false);
...
```

Reply:
Makes sense! Added them on b18db14